src/model/document.js

```diff
@@ -391,6 +396,28 @@ export default class Document {
 	 * @event change
 	 * @param {module:engine/model/batch~Batch} batch The batch that was used in the executed changes block.
 	 */
+
+	/**
+	 * Fired after each {@link module:engine/model/model~Model#enqueueChange `enqueueChange()` block},
+	 * which changes the data model or the outermost {@link module:engine/model/model~Model#change `change()` block}
+	 * was executed and the document has been changed during that block's execution.
+	 *
+	 * This event fires before the {@link module:engine/model/model~Model#change} event.
```

Comment:
Is this true? Aren't listeners for both these events combined together?

Reply:
If someone listens to both `change` and `change:data` events, the `change:data` will fire before the `change` event. Because each fired `change:...` event will fire the `change` event too.

Reply:
![image](https://user-images.githubusercontent.com/156149/41300115-4391be9a-6e65-11e8-8fb6-271c1380c9d1.png)

![image](https://user-images.githubusercontent.com/156149/41300165-5e7fc26a-6e65-11e8-9c2d-a168c857ef0b.png)

Only the order of registering these listeners matter.

So, as I wrote, the listeners are combined when the event is fired. Or, in other words, when you listen to `x:y` you also listen to `x`. But only one event is fired.

Reply:
Right, right, right. I was completely wrong. 👍